protocol/rpcprovider/rpcprovider.go

```diff
@@ -270,6 +285,43 @@ func ParseEndpoints(viper_endpoints *viper.Viper, geolocation uint64) (endpoints
 	return
 }
 
+func fetchVersionFromConsensus() (string, string, error) {
```

Comment:
not at all how we communicate with the blockchain through cmds... 

you were supposed to use the state_tracker and the event_tracker to implement the updater 
in NewStateTracker you can validate the current version and initialize see how we query other information there 

first initializing the current version in the NewStateTracker method and then listening to changes while implementing an updater (i.e spec_updater does the same for spec updates) 

it's like a service that waits for a specific action, also when identifying mismatching versions, (within the allowed range) instead of a warning print an error and do this for every block update (so its a bit spamming so the user will notice)